onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -381,15 +381,19 @@ def __init__(
 
         try:
             self._create_inference_session(providers, provider_options, disabled_optimizers)
-        except ValueError as e:
+        except (ValueError, RuntimeError) as e:
             if self._enable_fallback:
-                print(f"EP Error {e} when using {providers}")
-                print(f"Falling back to {self._fallback_providers} and retrying.")
-                self._create_inference_session(self._fallback_providers, None)
-                # Fallback only once.
-                self.disable_fallback()
-                return
-            raise
+                try:
+                    print(f"EP Error {e} when using {providers}")
+                    print(f"Falling back to {self._fallback_providers} and retrying.")
+                    self._create_inference_session(self._fallback_providers, None)
+                    # Fallback only once.
+                    self.disable_fallback()
+                    return
+                except Exception as fallback_error:
```

Comment:
What is the intent behind this try catch? If not caught, it would automatically be raised, right?

Reply:
It's for maintaining the source error which triggering this fallback. This can give the debug person a global picture of what's going on.